x-pack/plugins/cases/public/client/attachment_framework/types.ts

```diff
@@ -38,7 +39,9 @@ export interface AttachmentType<Props> {
   id: string;
   icon: IconType;
   displayName: string;
-  getAttachmentViewObject: () => AttachmentViewObject<Props>;
+  getAttachmentViewObject: (
+    props: ExternalReferenceAttachmentViewProps
```

Comment:
If other attachment types have a function that looks like `getAttachmentViewObject: () => AttachmentViewObject<props>` would this change cause an error? It doesn't look like it when I look at those other implementations. So I guess it's fine? @cnasikas ?

Reply:
I think it should be fine, we discussed it briefly offline before I created the PR.

Other functions get the props but just don't do anything with it. 🤷

Reply:
I think it is fine yes.

Reply:
@adcoelho The type of props should be generic as the framework will pass to both types of attachments the needed props. We should change it to `getAttachmentViewObject: (props: Props) => AttachmentViewObject<Props>;`